packages/evm/test/court/controller/controller-modules.js

```diff
@@ -616,4 +617,118 @@ contract('Controller', ([_, fundsGovernor, configGovernor, modulesGovernor, some
       })
     })
   })
+
+  describe('customFunctions', () => {
+    let module
+
+    const setCounterSig = sha3('setCounter(uint256)').slice(0, 10)
+    const receiveEtherSig = sha3('receiveEther()').slice(0, 10)
+    const failSig = sha3('fail()').slice(0, 10)
+
+    beforeEach('deploy module', async () => {
+      module = await ControlledMock.new(controller.address)
+    })
+
+    context('when the sender is the governor', () => {
+      const from = modulesGovernor
+
+      context('when setting a function', () => {
+        const itRegistersTheCustomFunction = () => {
+          it('sets the function', async () => {
+            const receipt = await controller.setCustomFunction(setCounterSig, module.address, { from })
+            assertAmountOfEvents(receipt, CONTROLLER_EVENTS.CUSTOM_FUNCTION_SET)
+            assertEvent(receipt, CONTROLLER_EVENTS.CUSTOM_FUNCTION_SET, { signature: setCounterSig, target: module.address })
+
+            const anotherReceipt = await controller.setCustomFunction(receiveEtherSig, module.address, { from })
+            assertAmountOfEvents(anotherReceipt, CONTROLLER_EVENTS.CUSTOM_FUNCTION_SET)
+            assertEvent(anotherReceipt, CONTROLLER_EVENTS.CUSTOM_FUNCTION_SET, { signature: setCounterSig, target: module.address })
+          })
+
+          it('can be called', async () => {
+            const data = module.contract.methods.setCounter(10).encodeABI()
+            await controller.setCustomFunction(setCounterSig, module.address, { from })
+
+            assertBn(await module.counter(), 0, 'counter does not match')
+            await controller.sendTransaction({ data })
+            assertBn(await module.counter(), 10, 'counter does not match')
+          })
+
+          it('handles eth transfers properly', async () => {
```

Comment:
Could we add a test where we send just ETH or an empty-data transaction? I would suspect `msg.sig` would be `0x00000000`.

Reply:
Addressed in 3e5d18a6eeb5c9e20f57eacea79ba35151820a98